internal/engine/compiler/impl_arm64.go

```diff
@@ -490,13 +490,27 @@ func (c *arm64Compiler) setLocationStack(newStack *runtimeValueLocationStack) {
 
 // compileBuiltinFunctionCheckExitCode implements compiler.compileBuiltinFunctionCheckExitCode for the arm64 architecture.
 func (c *arm64Compiler) compileBuiltinFunctionCheckExitCode() error {
-	if err := c.compileCallGoFunction(nativeCallStatusCodeCallBuiltInFunction, builtinFunctionIndexCheckExitCode); err != nil {
+	dstReg := arm64ReservedRegisterForTemporary
+
+	// dstReg := callEngine.moduleContext.moduleInstance
+	c.assembler.CompileMemoryToRegister(arm64.LDRD,
+		arm64ReservedRegisterForCallEngine, callEngineModuleContextModuleInstanceOffset,
+		dstReg)
+	// dstReg = *dstReg; i.e. (*(callEngine.moduleContext.moduleInstance)).Closed
+	c.assembler.CompileMemoryToRegister(arm64.LDRD, dstReg, 0, dstReg)
+	// If dstReg == 0 then we are not quitting, skip over all the following.
+	c.assembler.CompileTwoRegistersToNone(arm64.CMP, arm64.RegRZR, dstReg)
+	brIfZero := c.assembler.CompileJump(arm64.BCONDEQ)
+	if err := c.compileCallGoFunction(nativeCallStatusCodeCallBuiltInFunction, builtinFunctionIndexExitUnconditionally); err != nil {
 		return err
 	}
 
-	// After return, we re-initialize reserved registers just like preamble of functions.
-	c.compileReservedStackBasePointerRegisterInitialization()
-	c.compileReservedMemoryRegisterInitialization()
+	//// After return, we re-initialize reserved registers just like preamble of functions.
+	//c.compileReservedStackBasePointerRegisterInitialization()
+	//c.compileReservedMemoryRegisterInitialization()
```

Comment:
I think this is not necessary since we are exiting, thus we don't need to clean after ourselves

Reply:
yes can be deleted